### ISSUE TITLE
GalleryViewPager Crashed with IllegalArgumentException

### ISSUE THREAD

**Issue by jackcsk** [101medialab/android-loyalnativeslider#1]
Received Fabric crash report on GalleryViewPager crash:

```
Fatal Exception: java.lang.IllegalArgumentException: pointerIndex out of range
       at android.view.MotionEvent.nativeGetAxisValue(MotionEvent.java)
       at android.view.MotionEvent.getX(MotionEvent.java:2072)
       at android.support.v4.view.MotionEventCompatEclair.getX(MotionEventCompatEclair.java:32)
       at android.support.v4.view.MotionEventCompat$EclairMotionEventVersionImpl.getX(MotionEventCompat.java:110)
       at android.support.v4.view.MotionEventCompat.getX(MotionEventCompat.java:462)
       at android.support.v4.view.ViewPager.onInterceptTouchEvent(ViewPager.java:1992)
       at com.hkm.slider.GalleryWidget.GalleryViewPager.onInterceptTouchEvent(GalleryViewPager.java:128)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2108)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2559)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2260)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2559)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2260)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2559)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2260)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2559)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2260)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2559)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2260)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2559)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2260)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2559)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2260)
       at com.android.internal.policy.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:2439)
       at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1741)
       at android.app.Activity.dispatchTouchEvent(Activity.java:2776)
       at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:60)
       at com.android.internal.policy.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2388)
       at android.view.View.dispatchPointerEvent(View.java:9590)
       at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4424)
       at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4280)
       at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3804)
       at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3863)
       at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3829)
       at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3959)
       at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3837)
       at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4016)
       at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3809)
       at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3863)
       at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3829)
       at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3837)
       at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3809)
       at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6138)
       at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6106)
       at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6060)
       at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6241)
       at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:216)
       at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents(InputEventReceiver.java)
       at android.view.InputEventReceiver.consumeBatchedInputEvents(InputEventReceiver.java:202)
       at android.view.ViewRootImpl.doConsumeBatchedInput(ViewRootImpl.java:6212)
       at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run(ViewRootImpl.java:6264)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:873)
       at android.view.Choreographer.doCallbacks(Choreographer.java:676)
       at android.view.Choreographer.doFrame(Choreographer.java:600)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:859)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:168)
       at android.app.ActivityThread.main(ActivityThread.java:5781)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:797)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:687)
```

**Comment by jackcsk** [comment]
On other open source projects suffering the same issue, it is reported that using multi-finger gesture but ended with less finger could cause the crash. This symptom can be replicated with three finger swipe, but remove one finger in the middle of the gesture.

Related issues reports:
https://github.com/chrisbanes/PhotoView/issues/31

This issue is also reported on AOSP:
https://code.google.com/p/android/issues/detail?id=163954

**Comment by jackcsk** [comment]
Workaround: will implement the try-catch block as suggested:
http://stackoverflow.com/questions/16459196/java-lang-illegalargumentexception-pointerindex-out-of-range-exception-dispat

**Comment by jackcsk** [comment]
Resolved in commit 4efa3c4e0698cc7608006044b1ed8b37a6683cd8
